### PR TITLE
(webdriverio): don't fail getContext is not supported, e.g. using Tizen TV driver

### LINKED ISSUE #12196: [🐛 Bug]: Tizen TV platform erroneously calls getContext and crashes
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.22.1

### Node.js Version

18.18.2

### Mode

Standalone Mode

### Which capabilities are you using?

```typescript
{
    platformName: 'TizenTv',
    'appium:automationName': 'TizenTv',
    'appium:deviceName': '<TV IP>:26101',
    'appium:appPackage': '<Tizen app name>',
    'appium:chromedriverExecutable': '/Users/brwalsh/chromedriver',
    'appium:autoWebview': true
}
```

### What happened?

While running a webdriver test on a Samsung Tizen TV with appium, the test crashes and fails upon calling [isClickable](https://webdriver.io/docs/api/element/isClickable/). This is due to isClickable [calling getContext](https://github.com/webdriverio/webdriverio/blob/3d44631bdcb332238b29aa0859dae7b335e51f89/packages/webdriverio/src/commands/element/isClickable.ts#L49), which is undefined for the Tizen TV platform. The same issue is present in [isEqual](https://github.com/webdriverio/webdriverio/blob/3d44631bdcb332238b29aa0859dae7b335e51f89/packages/webdriverio/src/commands/element/isEqual.ts#L40). Both calls are gated behind checks of `browser.isMobile()`, but that returns true for Tizen TV.

### What is your expected behavior?

`isClickable` and `isEqual` complete as expected without running any platform-incompatible code.

### How to reproduce the bug.

Build the [tizen-sample-app](https://github.com/headspinio/appium-tizen-tv-driver/tree/main/packages/tizen-sample-app) from the appium-tizen-tv-driver repo and deploy it to a Samsung Tizen Smart TV. Install [Appium](https://appium.io/docs/en/latest/quickstart/install/) and the [Tizen TV driver](https://github.com/headspinio/appium-tizen-tv-driver/tree/main/packages/appium-tizen-tv-driver), then modify any [e2e test](https://github.com/headspinio/appium-tizen-tv-driver/blob/main/packages/appium-tizen-tv-driver/test/e2e/driver.e2e.spec.js) to call `isClickable` or `isEqual` on a page element. 

I am also happy to test fixes on your behalf with my TV setup if nobody on the dev team has access to a Tizen TV.

### Relevant log output

```typescript
[0-0] 2024-02-07T21:51:54.626Z INFO webdriver: COMMAND isElementDisplayed("4986e520-99e0-4acc-88b2-969d4d84ec4e")
[0-0] 2024-02-07T21:51:54.627Z INFO webdriver: [GET] http://127.0.0.1:4723/session/ccc56a56-ac05-4cd9-929e-ff68ff71d40c/element/4986e520-99e0-4acc-88b2-969d4d84ec4e/displayed
[0-0] 2024-02-07T21:51:54.725Z INFO webdriver: RESULT true
[0-0] 2024-02-07T21:51:54.725Z INFO webdriver: COMMAND getContext()
[0-0] 2024-02-07T21:51:54.725Z INFO webdriver: [GET] http://127.0.0.1:4723/session/ccc56a56-ac05-4cd9-929e-ff68ff71d40c/context
[0-0] 2024-02-07T21:51:54.733Z DEBUG webdriver: request failed due to response error: unknown command
[0-0] 2024-02-07T21:51:54.733Z WARN webdriver: Request failed with status 404 due to unknown command: unknown command: session/06894397273a76da1fbf30a7fca70819/context
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

fixes #12196

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

n/a

### CODE REVIEW

File: packages/webdriverio/src/commands/element/isClickable.ts [hunk 1]
```diff
@@ -46,7 +46,11 @@ export async function isClickable (this: WebdriverIO.Element) {
         return false
     }
 
-    if (this.isMobile && await this.getContext() === 'NATIVE_APP') {
+    /**
+     * some Appium platforms don't support the `getContext` method, in that case
+     * we can't determine if we are in a native context or not, so we return undefined
+     */
+    if (this.isMobile && await this.getContext().catch(() => undefined) === 'NATIVE_APP') {
```
Comment (wswebcreation) [inline packages/webdriverio/src/commands/element/isClickable.ts:53]: @christian-bromann 

Why are we handling this in the method? Why not make something re-usable for WDIO so we have the context and we can access it in all methods that are not usable for mobile

- [set it for 95% of the cases](https://github.com/webdriverio/visual-testing/blob/main/packages/service/src/service.ts#L50)
- [the method](https://github.com/webdriverio/visual-testing/blob/main/packages/service/src/utils.ts#L228)
- [command check](https://github.com/webdriverio/visual-testing/blob/main/packages/service/src/service.ts#L70)

Comment (christian-bromann) [reply packages/webdriverio/src/commands/element/isClickable.ts:53]: Mind raising an issue for this? This makes totally sense to extend the environment flag with something like `isNative`.

Comment (wswebcreation) [reply packages/webdriverio/src/commands/element/isClickable.ts:53]: Done, see https://github.com/webdriverio/webdriverio/issues/12208

Comment (christian-bromann) [reply packages/webdriverio/src/commands/element/isClickable.ts:53]: Thanks!

**Pull request merged**